### ISSUE TITLE
Hide navigation on small screens

### ISSUE THREAD

**Issue by korelstar** [nextcloud/appstore#432]
On small screens with a width lower than 768px (e.g. mobile phones), the navigation (category list) doesn't float on the left side but moves above of the content (responsive design with media queries). This approach is fine and it fits perfectly for the startpage. But if I've chosen an app to view, I have to scroll down the long list of categories, before I can see the desired content (cp. screenshot).

![screenshot](https://cloud.githubusercontent.com/assets/6277619/22627916/4619953e-ebcb-11e6-91e8-d10ea5d4e12a.png)

Therefore, I propose to change the appearance of the category list as following (in dependance of the type of page shown):

- **Start page** (`https://apps.nextcloud.com/`): keep the current behavior, i.e. show the category list always.
- **Category view** (`https://apps.nextcloud.com/categories/*`): collapse the category list and show only the current category selection; provide a button to expand the category list using JavaScript.
- **App view** (`https://apps.nextcloud.com/apps/*`): don't show the category list at all; you can go back to the list of categories by clicking the *Nextcloud App Store* icon on the top.

This behavior should apply only to the media query for small devices, i.e. it has to be realized in CSS. The layout for larger devices can be kept, in my view.

**Comment by BernhardPosselt** [comment]
Good point however I'd move everything into a js sub menu for simplicity and usibility purposes

**Comment by jancborchardt** [comment]
We can even just combine it in the existing menu, adding a "Categories" section with the categories as subitems.

**Comment by brokoli777** [comment]
Hi, I would like to work on this feature. Like @BernhardPosselt mentioned, I think a side menu would be best. I will try to show a mockup of it later

**Comment by brokoli777** [comment]
![image](https://github.com/user-attachments/assets/946e284d-8e78-42f8-af24-be0d1a81c509)
![Screenshot 2024-09-30 004445](https://github.com/user-attachments/assets/b7ef18a3-158b-4e81-a489-1ad51c384b6e)
